.agents/skills/typescript-expert/scripts/ts_diagnostic.py

```
@@ -0,0 +1,203 @@
+"""
+
+import subprocess
+import sys
+import os
+import json
+from pathlib import Path
+
+def run_cmd(cmd: str) -> str:
+    """Run shell command and return output."""
+    try:
+        result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
```

Ruff (0.15.1)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)